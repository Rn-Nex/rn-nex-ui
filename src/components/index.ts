export * from './Accordion';
export * from './ActivityIndicator';
export * from './Avatar';
export * from './Badge';
export * from './Box';
export * from './Button';
export * from './Card';
export * from './CheckBox';
export * from './Chip';
export * from './Dialog';
export * from './Divider';
export * from './DropDown';
export * from './Image';
export * from './List';
export * from './Pagination';
export * from './Portal';
export * from './Radio';
export * from './Snackbar';
export * from './Stepper';
export * from './Switch';
export * from './TextField';
export * from './Transitions';
export * from './Typography';
