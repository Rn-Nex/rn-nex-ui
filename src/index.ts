export * from './components';
export * from './libraries';
export * from './utils';
