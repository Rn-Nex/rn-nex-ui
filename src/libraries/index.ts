export * from './themes/v2';
export * from './style';
