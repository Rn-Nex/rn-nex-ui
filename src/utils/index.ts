export * from './styleGenerator';
export * from './randomId';
export * from './utils';
export * from './dimensions';
