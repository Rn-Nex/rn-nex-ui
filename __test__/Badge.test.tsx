import { render } from '@testing-library/react-native';
import React from 'react';
import { View } from 'react-native';
import { Badge, BADGE_DEFAULT_RADIUS, green, lightBlue, primary, red, secondary, ThemeProvider, yellow } from '../src';

describe('Badge', () => {
  const mockBadgeTestId = 'mock-badge-test-id';
  const mockRef = React.createRef<View>();

  beforeAll(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Badge />
      </ThemeProvider>,
    );
    expect(toJSON).toMatchSnapshot();
  });

  it('should render correctly with inner component', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Badge>
          <View />
        </Badge>
      </ThemeProvider>,
    );
    expect(toJSON).toMatchSnapshot();
  });

  it('should forward the ref', () => {
    render(
      <ThemeProvider>
        <Badge ref={mockRef} />
      </ThemeProvider>,
    );
    expect(mockRef.current).not.toBeNull();
    expect(mockRef.current).toBeInstanceOf(View);
  });

  it('should render the badge content', () => {
    const { getByText } = render(
      <ThemeProvider>
        <Badge badgeContent={'100'} max={1000} />
      </ThemeProvider>,
    );

    const label = getByText('100');
    expect(label).toBeDefined();
  });

  it('should render the primary variation of the badge component', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <Badge testID={mockBadgeTestId} variation="primary" />
      </ThemeProvider>,
    );

    const badge = getByTestId(mockBadgeTestId);
    expect(badge.props.style).toEqual(expect.objectContaining({ backgroundColor: primary[500] }));
  });

  it('should render the secondary variation of the badge component', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <Badge testID={mockBadgeTestId} variation="secondary" />
      </ThemeProvider>,
    );

    const badge = getByTestId(mockBadgeTestId);
    expect(badge.props.style).toEqual(expect.objectContaining({ backgroundColor: secondary[500] }));
  });

  it('should render the error variation of the badge component', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <Badge testID={mockBadgeTestId} variation="error" />
      </ThemeProvider>,
    );

    const badge = getByTestId(mockBadgeTestId);
    expect(badge.props.style).toEqual(expect.objectContaining({ backgroundColor: red[500] }));
  });

  it('should render the info variation of the badge component', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <Badge testID={mockBadgeTestId} variation="info" />
      </ThemeProvider>,
    );

    const badge = getByTestId(mockBadgeTestId);
    expect(badge.props.style).toEqual(expect.objectContaining({ backgroundColor: lightBlue[500] }));
  });

  it('should render the success variation of the badge component', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <Badge testID={mockBadgeTestId} variation="success" />
      </ThemeProvider>,
    );

    const badge = getByTestId(mockBadgeTestId);
    expect(badge.props.style).toEqual(expect.objectContaining({ backgroundColor: green[500] }));
  });

  it('should render the warning variation of the badge component', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <Badge testID={mockBadgeTestId} variation="warning" />
      </ThemeProvider>,
    );

    const badge = getByTestId(mockBadgeTestId);
    expect(badge.props.style).toEqual(expect.objectContaining({ backgroundColor: yellow[500] }));
  });

  it('should render the badge dot component', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <Badge testID={mockBadgeTestId} variant="dot" />
      </ThemeProvider>,
    );
    const badge = getByTestId(mockBadgeTestId);
    expect(badge.props.style).toEqual(
      expect.objectContaining({ minWidth: BADGE_DEFAULT_RADIUS, minHeight: BADGE_DEFAULT_RADIUS }),
    );
  });
});
