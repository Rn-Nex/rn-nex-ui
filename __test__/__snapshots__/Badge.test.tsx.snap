// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render correctly 1`] = `[Function]`;

exports[`Badge should render correctly with inner component 1`] = `[Function]`;
